components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandlerTest.java

```diff
@@ -205,6 +236,144 @@ public void testBuildIntrospectionResponse(boolean isIDPIdColumnEnabled, String
+                                                List<String> parentHierarchyFromSwitchedOrg,
+                                                String subOrgStartLevel,
+                                                boolean expectedValidatorResponse)
+            throws Exception {
+
+        mockRequiredObjects();
+
+        OAuth2TokenValidationRequestDTO oAuth2TokenValidationRequestDTO = new OAuth2TokenValidationRequestDTO();
+        OAuth2TokenValidationRequestDTO.OAuth2AccessToken oAuth2AccessToken = oAuth2TokenValidationRequestDTO.new
+                OAuth2AccessToken();
+        oAuth2AccessToken.setIdentifier(jwtToken);
+        oAuth2AccessToken.setTokenType("bearer");
+        oAuth2TokenValidationRequestDTO.setAccessToken(oAuth2AccessToken);
+
+        OAuth2TokenValidationRequestDTO.TokenValidationContextParam tokenValidationContextParam =
+                mock(OAuth2TokenValidationRequestDTO.TokenValidationContextParam.class);
+        tokenValidationContextParam.setKey("dummy");
+        tokenValidationContextParam.setValue("dummy");
+        OAuth2TokenValidationRequestDTO.TokenValidationContextParam[] tokenValidationContextParams =
+                {tokenValidationContextParam};
+        oAuth2TokenValidationRequestDTO.setContext(tokenValidationContextParams);
+
+        OAuth2TokenValidationResponseDTO oAuth2TokenValidationResponseDTO = new OAuth2TokenValidationResponseDTO();
+
+        OAuth2TokenValidationMessageContext validationReqDTO = new OAuth2TokenValidationMessageContext(
+                oAuth2TokenValidationRequestDTO, oAuth2TokenValidationResponseDTO);
+
+        authzUser.setTenantDomain(clientAppTenantDomain);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(clientAppTenantDomain);
+
+        mockStatic(IdentityProviderManager.class);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+        when(identityProviderManager.getResidentIdP(Mockito.anyString())).thenReturn(identityProvider);
+        stub(method(OAuth2JWTTokenValidator.class, "getSigningTenantDomain", JWTClaimsSet.class, AccessTokenDO.class))
+                .toReturn(clientAppTenantDomain);
+
+        FederatedAuthenticatorConfig[] federatedAuthenticatorConfigs = new FederatedAuthenticatorConfig[0];
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(federatedAuthenticatorConfigs);
+        mockStatic(IdentityApplicationManagementUtil.class);
+        mockStatic(FederatedAuthenticatorConfig.class);
+        when(IdentityApplicationManagementUtil.getFederatedAuthenticator(federatedAuthenticatorConfigs,
+                "openidconnect")).thenReturn(federatedAuthenticatorConfig);
+
+        Property[] properties = new Property[0];
+        Property property = new Property();
+        property.setName("IdPEntityId");
+        property.setValue("https://api.asg.io/o/" + switchedOrganizationId + "/oauth2/token");
```

Comment:
use localhost url

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2086/commits/c86d29ea220afa80a0b2cde20edce415b4d6a938